solutions/src/test/java/ru/sberbank/school/task10/ScalableThreadPoolTest.java

```diff
@@ -1,65 +1,46 @@
 package ru.sberbank.school.task10;
 
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Random;
 import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Future;
+import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 
 class ScalableThreadPoolTest {
 
     @Test
-    void workingTestRunnable() throws InterruptedException {
-        ScalableThreadPool pool = new ScalableThreadPool(10, 20);
+    void assertThatTemporaryThreadsWasCreatedAndTerminated() throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(120);
+        ScalableThreadPool pool = new ScalableThreadPool(3, 10);
         pool.start();
-        for (int i = 0; i < 100; i++) {
+        for (int i = 0; i < 120; i++) {
+            TimeUnit.MILLISECONDS.sleep((long) (10 * Math.log(i)));
             pool.execute(() -> {
                 try {
-                    System.out.println("Thread " + Thread.currentThread().getName() + " is working");
-                    TimeUnit.SECONDS.sleep(1);
-                } catch (InterruptedException e) {
-                    e.printStackTrace();
-                }
+                    int timeout = new Random().nextInt(3);
+                    TimeUnit.SECONDS.sleep(timeout);
+                    latch.countDown();
+                } catch (InterruptedException ignored) {}
```

Comment:
Empty catch block.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a998603812>*